bin.src/genDefectList.py

```diff
@@ -0,0 +1,265 @@
+    medianFlatsLow = allFlatsLow[:,:,0]
+
+nHigh = len(flatsHigh)
+if nHigh > 2:
+    medianFlatsHigh = np.median(allFlatsHigh,2)
+    logger.info('%d Flats with %ds exposure time found' % (nLow, args.expTimeHigh))
+elif nHigh == 2:
+    medianFlatsHigh = np.min(allFlatsHigh,2)
+    logger.warn('Only %d Flats with %ds exposure time found' % (nLow, args.expTimeHigh))
+elif nHigh < 1:
+    raise RunTimeError("No flats with %d s exposure time found" % args.expTimeHigh)
+else:
+    logger.warn('Only 1 Flat with %ds exposure time' % args.expTimeHigh)
+    medianFlatsHigh = allFlatsHigh[:,:,0]
+
+medianFlatsLow = drpStella.where( medianFlatsLow, '==', 0., 0.000001, medianFlatsLow)
+medianFlatsHigh = drpStella.where( medianFlatsHigh, '==', 0., 0.000001, medianFlatsHigh)
+divFlat = medianFlatsHigh / medianFlatsLow
+
+sctrl = afwMath.StatisticsControl()
+sctrl.setWeighted(False)
+
+divFlatIm = afwImage.ImageF(divFlat)
+divFlatMIm = afwImage.makeMaskedImage(divFlatIm)
+
+mean = afwMath.makeStatistics(divFlatMIm, afwMath.MEANCLIP).getValue()
+stddev = afwMath.makeStatistics(divFlatMIm,afwMath.STDEVCLIP).getValue()
+
+if args.outFile != '':
+    mask = divFlatMIm.getMask()
+    maskVal = 1 << divFlatMIm.getMask().getMaskPlane('BAD')
+    for badCol in badCols:
+        text_file.write("%d %d 0 1 %d\n" % (ccd, badCol, args.nRows))
+        mask[badCol,:] |= maskVal
+    nBad = 0
+    for iRow in np.arange(args.nPixCut,divFlat.shape[0]-args.nPixCut):
+        for iCol in np.arange(args.nPixCut,args.gapLow):
+            if iCol not in badCols:
+                if np.absolute(divFlat[iRow, iCol] - mean) > args.maxSigma * stddev:
+                    nBad = nBad + 1
+                    text_file.write("%d %d %d 1 1\n" % (ccd, iCol, iRow))
+                    mask[iCol, iRow] |= maskVal;
+
+    for iRow in np.arange(args.nPixCut,divFlat.shape[0]-args.nPixCut):
+        for iCol in np.arange(args.gapHigh,args.nCols-args.nPixCut):
+            if iCol not in badCols:
+                if np.absolute(divFlat[iRow, iCol] - mean) > args.maxSigma * stddev:
+                    nBad = nBad + 1
+                    text_file.write("%d %d %d 1 1\n" % (ccd, iCol, iRow))
+                    mask[iCol, iRow] |= maskVal;
+
+    if textArrRead:
+        for a in uniqueFirstIntsSorted:
+            if a > ccd:
+                for i in range(len(textArr)):
+                    if firstInts[i] == a:
+                        text_file.write(textArr[i])
+    text_file.write("# Dead amps")
+    text_file.close()
+
+    logger.info('%d bad pixels found' % nBad)
+
+if args.medianFlatsOut != '' or args.display:
+    medianFlatsLowExp = afwImage.makeExposure(afwImage.makeMaskedImage(afwImage.ImageF(medianFlatsLow)))
+    medianFlatsLowExp.getMaskedImage().getMask()[:,:] = mask[:,:]
+    medianFlatsHighExp = afwImage.makeExposure(afwImage.makeMaskedImage(afwImage.ImageF(medianFlatsHigh)))
+    medianFlatsHighExp.getMaskedImage().getMask()[:,:] = mask[:,:]
```

Comment:
Added function getMedianFlats which returns a dict containing the median flats for the shorter and larger exposure times respectively, and the quotient